jax/_src/dispatch.py

```diff
@@ -430,12 +416,10 @@ def jaxpr_has_pmap(jaxpr):
   return False
 
 def jaxpr_has_bints(jaxpr: core.Jaxpr) -> bool:
-  return (any(type(d) is core.Var for v in jaxpr.invars
-              if type(v.aval) is core.DShapedArray for d in v.aval.shape) or
-          any(type(d) is core.Var
+  return (any(type(v.aval) is core.AbstractBInt for v in jaxpr.invars) or
```

Comment:
This check was buggy before. The logic is that we want to check all binders' annotations (i.e. the lambda binders in `jaxpr.invars` and the let binders in each `eqn.outvars`) to see if there are any `AbstractBInt`s, recursing into subjaxprs. That checks every in-scope variable.